src/stampit.js

```diff
@@ -120,15 +120,60 @@ const stampit = function stampit(options) {
     let instance = mixin(create(fixed.methods), fixed.refs, refs);
     mergeUnique(instance, fixed.props); // props are safely merged into refs
 
+    let nextPromise = null;
     if (fixed.init.length > 0) {
       forEach(fixed.init, fn => {
-        if (isFunction(fn)) {
-          instance = fn.call(instance, { args, instance, stamp: factory }) || instance;
+        if (!isFunction(fn)) {
+          return; // not a function, do nothing.
+        }
+
+        // Check if we are in the async mode.
+        if (!nextPromise) {
+          // Call the init().
+          const callResult = fn.call(instance, {args, instance, stamp: factory});
+          if (!callResult) {
+            return; // nothing got returned. Proceed.
+          }
+
+          // Returned value is meaningful. It will replace the stampit-created object.
+          if (!isThenable(callResult)) {
+            instance = callResult; // stamp is synchronous so far.
+            return;
+          }
+
+          // This is the sync->async conversion point. Since now our factory will return a promise, not an object.
+          nextPromise = callResult;
+        } else {
+          // As long as one of the init() functions returned a promise, now our factory will 100% return promise too.
+          // Linking the init() functions into the promise chain.
+          nextPromise = nextPromise.then((newInstance) => {
```

Comment:
We need to stay consistent with our arrow function formatting. The rest of the library excludes parenthesis around single parameters.

(newInstance) -> newInstance

This PR has a few other places doing this as well.